boazpoolman/strapi-plugin-config-sync
# Don't recompile typescript project if already compiled

<!--

### What does it do?

Avoid needlessly rebuilding the admin UI when running `yarn config-sync import `if already done.

### Why is it needed?

When running software in an image in Kubernetes or Docker, it's good practice to [have the image file system be read-only](https://www.thorsten-hans.com/read-only-filesystems-in-docker-and-kubernetes/). 

This means building the Strapi app in the `Dockerfile` at build time, but still running `config-sync import` at runtime.

### How to test it?

1. Use this branch of config-sync in a Strapi project
2. Run `config-sync import` in two scenarios:
    a. in a project that already has been built
    b. in a project that has not been built (`rm -rf dist`)
3. Verify that the expected config has been imported in both scenarios

I was unable to test it as per the instructions in CONTRIBUTING.md, because yarn kept complaining that `yarn config-sync` could not be found, despite it being in strapi-plugin-config-sync's `bin` folder.

---

Please let me know if I missed anything.
